Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs

```diff
@@ -410,12 +415,6 @@ internal static async Task DownloadOneFileAsync(SFFileMetadata fileMetadata, Can
             }
 
             ISFRemoteStorageClient client = fileMetadata.client;
-            FileHeader fileHeader = await client.GetFileHeaderAsync(fileMetadata, cancellationToken).ConfigureAwait(false);
```

Comment:
Moved the logic to [line 442](https://github.com/snowflakedb/snowflake-connector-net/pull/773/files#diff-141fae8d9ec0879939309639fad147fd50b1bba1902c0dbd13fd25fb5c10e3c7R442) to copy the non-async version of the function: https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/FileTransfer/StorageClient/SFRemoteStorageUtil.cs#L353